addons/base_iban/models/res_partner_bank.py

```diff
@@ -35,7 +35,7 @@ def validate_iban(iban):
         raise ValidationError(_("The IBAN is invalid, it should begin with the country code"))
 
     iban_template = _map_iban_template[country_code]
-    if len(iban) != len(iban_template.replace(' ', '')):
+    if len(iban) != len(iban_template.replace(' ', '')) or not re.fullmatch("[a-zA-Z0-9]+", iban):
```

Comment:
Hi @elre-odoo we actually have just one comment: could we use str.isalnum() instead?
https://docs.python.org/3/library/stdtypes.html#str.isalnum

Reply:
Hi @antoine162, thanks for the review!

Unfortunately, str.isalnum() would still allow non-ASCII alphanumeric characters such as `ü`. Another candidate could've been [str.isascii()](https://docs.python.org/3/library/stdtypes.html?highlight=isascii#str.isascii) (since the string has already been cleared of non-alphanumeric characters, this would only include the pattern `[a-zA-Z0-9]`), however this is only available since Python 3.7, i.e. Odoo 15+.

A more concise alternative would be to use the [re.ASCII](https://docs.python.org/3/library/re.html#re.ASCII) flag on the `[\W_]` pattern of re.sub() in [normalize_iban](https://github.com/odoo/odoo/blob/8a76f731c977852bbe4c1330a8a9d9059d79f777/addons/base_iban/models/res_partner_bank.py#L10); as noted in https://docs.python.org/3/library/re.html:
>`\W`
>    Matches any character which is not a word character. This is the opposite of `\w`. If the ASCII flag is used this becomes the equivalent of [^a-zA-Z0-9_].

With that, the `[\W_]` pattern would remove everything that's not `[a-zA-Z0-9]`, i.e. the pattern we need to check the IBAN for.

The difference to the existing proposal would be that the non-ASCII characters would be _removed_ before the IBAN validation and so a non-ASCII "IBAN" could theoretically still get validated; the existing proposal throws an error on any non-ASCII character and thus rejects any non-ASCII "IBAN" to begin with. So the choice is as follows:
- If we want to make sure that no string with non-ASCII characters could get treated as an IBAN, we should prefer the original suggestion and raise a ValidationError.
- If we want to be more lenient, and allow for a string with non-ASCII characters to be treated as an IBAN provided the non-ASCII parts are removed, we should prefer the solution with the ASCII flag in normalize_iban().

Reply:
Hi @elre-odoo oh wow I hadn't realized the difference between _alphanumeric_ and _ASCII_, thanks so much for explaining, and sorry for not understanding this subtlety at first!

It makes sense to just validate valid IBANs as does your original solution :smile:

Reply:
No worries, I had to look it up first as well :). Thanks for the review!